src/tests/src/libs/antares/study/short-term-storage-input/short-term-storage-input.cpp

```diff
@@ -274,4 +274,87 @@ BOOST_AUTO_TEST_CASE(check_container_properties_empty_file)
     removeIniFile();
 }
 
+BOOST_AUTO_TEST_CASE(check_series_lower_curve)
+{
+    createFileSeries(0.5, 8760);
+
+    BOOST_CHECK(series.loadFromFolder(folder));
+    BOOST_CHECK(series.validate());
+
+    series.lowerRuleCurve[120] = 0.6;
+
+    BOOST_CHECK(!series.validateCycle(100, std::optional<double>(0.5), 20));
+}
+
+BOOST_AUTO_TEST_CASE(check_series_upper_curve)
+{
+    createFileSeries(0.5, 8760);
+
+    BOOST_CHECK(series.loadFromFolder(folder));
+    BOOST_CHECK(series.validate());
+    //624 is third week first cycle (100 + 168*3 + 20)
+    series.upperRuleCurve[190] = 0.2;
+
+    BOOST_CHECK(!series.validateCycle(100, std::optional<double>(0.5), 45));
```

Comment:
```suggestion
    BOOST_CHECK(!series.validateCycle(100, 0.5, 45));
```
`std::optional<T>` has a conversion operator from `T`, so you don't need to explicitly specify it.